### PR TITLE
spec(core): ldml: Drop padding 🙀

### PR DESCRIPTION
**Pull request by srl295**
spec:
- only guarantee is 32 bit alignment
- also note a cross ref TODO for #7533

implementation:
- also add some checking in the builder
- add a Constants.str_section_id() for debugging
- update basic.txt

Note:
- basic.kmx went from 1,442 down to 1,366 bytes.

Fixes: #7987

@keymanapp-test-bot skip

### CODE REVIEW

File: core/src/ldml/C7043_ldml.md [hunk 1]
```diff
@@ -27,10 +27,7 @@ Draft spec PR: <https://github.com/unicode-org/cldr/pull/1847>
 
 - Data is divided into several sections. The very first section is the `'sect'`
   section which is the table of contents.
-- All sections, including the first, begin on a 128-bit boundary with zero
-  padding as needed.
-- All variable length sections have the variable part begin on a 128-bit
-  boundary, with zero padding as needed.
+- All sections, including the first, are padded to a 32-bit boundary.
```
Comment (darcywong00) [inline core/src/ldml/C7043_ldml.md:30]: There's no more padding, right?

Comment (srl295) [reply core/src/ldml/C7043_ldml.md:30]: > There's no more padding, right?

Right. It should say aligned.

Comment (srl295) [reply core/src/ldml/C7043_ldml.md:30]: > There's no more padding, right?

There's no padding but everything is on a 32 bit boundary

**Review by darcywong00 — APPROVED** [review]
lgtm

**Review by mcdurdin — APPROVED** [review]

**Pull request merged**